File: web/e2e/fixtures/dragonTigerFlow.ts
// Synthetic, deterministic fixture. No upstream payloads or credentials.
const dates = ['2026-09-15', '2026-09-16', '2026-09-17', '2026-09-18', '2026-09-21'];
const names = ['半导体', '机器人', '算力'];
const evidence = dates.flatMap(trade_date => Array.from({ length: 16 }, (_, i) => {
  const concepts = ['半导体', i % 2 ? '机器人' : '算力'];
  const original_net_value = (8 - i) * 1e8;
  return concepts.map(name => ({
    symbol: `${600001 + i}.SH`, name: `测试股票${i + 1}`, trade_date, range_days: 1,
    concepts, concept_id: `concept-${names.indexOf(name)}`, concept_name: name, allocation_count: 2,
    original_net_value, stock_net_value: original_net_value, net_value: original_net_value / 2,
    buy_value: 6e8, sell_value: (12e8 - original_net_value) / 2,
    org_net_value: original_net_value / 10, hot_money_net_value: original_net_value * .15,
  }));
}).flat());
const concepts = names.map((name, i) => {
  const rows = evidence.filter(r => r.concept_name === name);
  const daily = rows.filter(r => r.trade_date === dates[0]).reduce((s, r) => s + r.net_value, 0);
  return { id: `concept-${i}`, name, net_value: daily * 5, org_net_value: daily,
    hot_money_net_value: daily * 1.5, stock_count: new Set(rows.map(r => r.symbol)).size,
    values: dates.map((_, j) => daily * (j + 1)) };
}).sort((a, b) => b.net_value - a.net_value);
const raw = {
  version: 'dragon-tiger-flow-v1', revision: 'fixture', board: 'all', range_days: 1,
  trade_date: dates.at(-1)!, expected_trade_date: dates.at(-1)!, dates, complete: true, missing_dates: [],
  excluded_undisclosed_count: 0,
  summary: { stock_count: 16, net_value: 4e9, buy_value: 96e9, sell_value: 92e9,
    org_net_value: 8e8, hot_money_net_value: 12e8, top5_concentration: 1 },
  concepts, evidence, hot_money_details: [],
  source_quality: dates.map(trade_date => ({ trade_date, generated_at: `${trade_date}T11:30:00Z`, errors: {}, sources: { all: {}, org: {}, hot_money: {} } })),
  attribution: 'independent_overlapping_categories', source: 'fuyao:/api/a-share/special-data/dragon-tiger-list', unit: 'CNY',
};
export default raw;

File: src/finance_analysis/interfaces/api/v1/schemas/dragon_tiger_flow.py
"""Explicit public amounts; normalized raw payloads and credentials are never exposed."""

from datetime import date, datetime
from typing import Any, Literal
from pydantic import BaseModel, Field, model_validator

Board = Literal["all", "org", "hot_money"]


class Amounts(BaseModel):
    net_value: float | None
    buy_value: float | None
    sell_value: float | None
    org_net_value: float | None
    hot_money_net_value: float | None


class Summary(Amounts):
    stock_count: int
    top5_concentration: float | None


class Concept(BaseModel):
    id: str
    name: str
    net_value: float | None
    org_net_value: float | None
    hot_money_net_value: float | None
    stock_count: int
    values: list[float | None]


class Evidence(Amounts):
    symbol: str
    name: str
    trade_date: date
    range_days: Literal[1, 3]
    concepts: list[str]
    concept_id: str
    concept_name: str
    allocation_count: int
    original_net_value: float | None
    stock_net_value: float | None


class HotMoneyDetail(BaseModel):
    symbol: str
    name: str
    trade_date: date
    range_days: Literal[1, 3]
    hot_money_name: str
    hot_money_item_net_value: float | None


class Overview(BaseModel):
    version: str
    revision: str
    board: Board
    range_days: Literal[1, 3]
    trade_date: date | None
    expected_trade_date: date | None = None
    dates: list[date]
    complete: bool
    missing_dates: list[date]
    excluded_undisclosed_count: int
    summary: Summary
    concepts: list[Concept]
    evidence: list[Evidence]
    hot_money_details: list[HotMoneyDetail]
    source_quality: list[dict[str, Any]]
    attribution: str
    source: str
    unit: str


class DateStatus(BaseModel):
    trade_date: date
    generated_at: datetime
    boards: list[Board]
    errors: dict[str, str]


class StockContribution(BaseModel):
    symbol: str
    name: str
    net_value: float | None
    rows: list[Evidence]


class ConceptDetail(BaseModel):
    revision: str
    concept: Concept
    stocks: list[StockContribution]


class StockDetail(BaseModel):
    revision: str
    rows: list[Evidence]
    hot_money_details: list[HotMoneyDetail]


class RunRequest(BaseModel):
    trade_date: date | None = None
    backfill_days: int | None = Field(None, ge=1, le=31)
    missing_only: bool = True

    @model_validator(mode="after")
    def exclusive(self):
        if self.trade_date is not None and self.backfill_days is not None:
            raise ValueError("trade_date 与 backfill_days 不能同时提供")
        return self


class RunResponse(BaseModel):
    task_id: str
    status: str = "pending"
